File: svr_rm.R
# install.packages('e1071', repos = "http://cran.us.r-project.org")
library(ggplot2)
library(e1071)

dataset = read.csv('/Users/rmaffesoli/Documents/personal/ML_A_Z/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/Position_Salaries.csv')

regressor = svm(formula=Salary ~ .,
                data=dataset,
                type='eps-regression')

# y_pred = predict(regressor, data.frame(Level = 6.5))


ggplot() +
  geom_point(aes(x=dataset$Level, y = dataset$Salary),
                 color='red') +
  geom_line(aes(x=dataset$Level, y = predict(regressor, newdata=dataset)),
                 color='blue') +
  ggtitle('Truth or Bluff (Linear Regression)') +
  xlab('Level') +
  ylab('Salary')

File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/svr_rm.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.svm import SVR
from sklearn.preprocessing import StandardScaler

dataset = pd.read_csv('/Users/rmaffesoli/Documents/personal/ML_A_Z/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2:].values

sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y)

regressor = SVR(kernel='rbf')
regressor.fit(X, y)

X_pred = sc_X.transform(np.array([[6.5]]))
y_pred = regressor.predict(X_pred)
y_pred_it = sc_y.inverse_transform(y_pred)
y_it = sc_y.inverse_transform(y)

plt.scatter(X, y_it, color='red')
plt.scatter(X_pred, y_pred_it, color='green')
plt.plot(X, sc_y.inverse_transform(regressor.predict(X)), color='blue')
plt.title('Truth or Bluff (SVR)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()
